src/test/Keno.t.sol

```diff
@@ -44,4 +113,157 @@ contract KenoTransactions is KenoTest {
+
+        try keno.executeEntropyForTest(2, 2758876867868697) {
+            fail();
+        } catch Error(string memory error) {
+            assertEq(error, "round has ended.");
+        }
+    }
+
+    function testBet() public {
+        payable(address(alice)).transfer(100 ether);
+
+        uint256[] memory betInfo = new uint256[](4);
+        for (uint256 i = 0;  i < 4; i++) {
+            betInfo[i] = i + 1;
+        }
+
+        uint256 cur_balance = address(keno).balance;
+        alice.play(2, betInfo, 1 ether);
+        assertEq(address(keno).balance, cur_balance + 1 ether);
+    }
+
+
+    function testMultiBet() public {
+        payable(address(keno)).transfer(10000 ether);
+        payable(address(alice)).transfer(2000 ether);
+
+        uint256 balance = address(keno).balance;
+
+        for (uint256 j = 0;  j < 1000; j++) {
+            uint256[] memory betInfo = new uint256[](1);
+            betInfo[0] = j % 80 + 1;
+
+            alice.play(3, betInfo, 1 ether);
+        }
+
+        assertEq(address(keno).balance, 1000 ether + balance);
+    }
+
+
+    function testPayout() public {
+        payable(address(alice)).transfer(100 ether);
+
+        uint256[] memory betInfo = new uint256[](3);
+        betInfo[0] = 30;
+        betInfo[1] = 48;
+        betInfo[2] = 26;
+
+        alice.play(2, betInfo, 1 ether);
+        hevm.roll(2);
+        keno.executeEntropyForTest(2, 2758876867868697);
+
+        assertEq(address(alice).balance, 124 ether);
+    }
+
+
+    function testMultiPlayerPayout() public{
+
+        uint256 vault_before = address(keno).balance;
+        uint256 vault_expect = vault_before;
+
+        uint256 testCount = 40;
+        User[] memory testUsers = new User[](testCount);
+
+        for(uint256 i = 0; i < testCount; i++) {
+            User ano = new User(payable(keno));
+            payable(address(ano)).transfer(2 ether);
+
+            testUsers[i] = ano;
+
+            uint256[] memory betInfo = new uint256[](3);
+            betInfo[0] = 30;
+            betInfo[1] = 48;
+            betInfo[2] = 26;
+
+            ano.play(2, betInfo, 1 ether);
+            vault_expect = vault_expect + 1 ether;
+        }
+
+        hevm.roll(2);
+        keno.executeEntropyForTest(2, 2758876867868697);
+
+        for (uint256 j = 0; j < testCount; j++) {
+            vault_expect = vault_expect - 25 ether;
+            assertEq(address(testUsers[j]).balance, 26 ether);
+        }
+
+        assertEq(address(keno).balance, vault_expect);
+    }
+
+    function testExceedPayout() public {
+
+        payable(address(alice)).transfer(100 ether);
+        uint256 keno_balance = address(keno).balance;
+        uint256 bet_amount = keno_balance / 25 + 2 ether;
+
+        uint256[] memory betInfo = new uint256[](3);
+        betInfo[0] = 30;
+        betInfo[1] = 48;
+        betInfo[2] = 26;
+
+        try alice.play(3, betInfo, bet_amount + 1 ether) {
+            emit log("test exceed payout failed");
```

Comment:
what is this for?

Reply:
just test the case at when the maxPayout exceed the vault of keno, but it's not correct here, I modify it at next commit